macros/supporting/hash.sql

```diff
@@ -9,14 +9,16 @@
     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
     See the License for the specific language governing permissions and
     limitations under the License.
--#}
+#}
+
 
-{%- macro _snowflake_standardize_hash_input(expr) -%}
+{%- macro _standardize_hash_input(expr, output_str_if_null='^^') -%}
 {#-
     Casts any field into a standardized string to prepare for hashing
 
     Arguments:
         expr {str} - the expression to standardize for hashing
+        output_str_if_null {str} - the string to output if the input expression is null (default: '^^')
```

Comment:
I know you're still thinking of a way to deal with '^^' and'||' config , but I included this here for now anyway.